esphome/components/exposure_notifications/exposure_notifications.cpp

```diff
@@ -0,0 +1,45 @@
+#include "exposure_notifications.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+
+#ifdef ARDUINO_ARCH_ESP32
+
+namespace esphome {
+namespace exposure_notifications {
+
+using namespace esp32_ble_tracker;
+
+static const char *TAG = "exposure_notifications";
+
+bool ExposureNotificationTrigger::parse_device(const ESPBTDevice &device) {
+  // See also https://blog.google/documents/70/Exposure_Notification_-_Bluetooth_Specification_v1.2.2.pdf
+  if (device.get_service_uuids().size() != 1)
+    return false;
+
+  // Exposure notifications have Service UUID FD 6F
+  ESPBTUUID uuid = device.get_service_uuids()[0];
+  if (!(uuid == ESPBTUUID::from_uint16(0xFD6F)))
+    return false;
+  if (device.get_service_datas().size() != 1)
+    return false;
+
+  // The service data should be 20 bytes
+  // First 16 bytes are the rolling proximity identifier (RPI)
+  // Then 4 bytes of encrypted metadata follow which can be used to get the transmit power level.
+  ServiceData service_data = device.get_service_datas()[0];
+  auto data = service_data.data;
```

Comment:
`ServiceData` is never used again, why not just merge the two statements?
```suggestion
  auto data = device.get_service_datas()[0].data;
```

Reply:
Reason was initially because the service data field also has a constant "magic" uuid that needs to be checked.

But apparently I forgot to check that.